src/Sema.zig

```diff
@@ -32334,7 +32334,10 @@ fn coerceTupleToStruct(
                 anon_struct_type.names.get(ip)[tuple_field_index]
             else
                 try ip.getOrPutStringFmt(sema.gpa, "{d}", .{tuple_field_index}, .no_embedded_nulls),
-            .struct_type => ip.loadStructType(inst_ty.toIntern()).field_names.get(ip)[tuple_field_index],
+            .struct_type => if (ip.loadStructType(inst_ty.toIntern()).field_names.len > 0)
+                ip.loadStructType(inst_ty.toIntern()).field_names.get(ip)[tuple_field_index]
+            else
+                try ip.getOrPutStringFmt(sema.gpa, "{d}", .{tuple_field_index}, .no_embedded_nulls),
```

Comment:
```suggestion
            .struct_type => ip.loadStructType(inst_ty.toIntern()).fieldName(ip, tuple_field_index).unwrap() orelse
                try ip.getOrPutStringFmt(sema.gpa, "{d}", .{tuple_field_index}, .no_embedded_nulls),
```

Reply:
Although, really it should probably be more like:
```zig
	const maybe_field_name = switch (ip.indexToKey(inst_ty.toIntern())) {
	   .anon_struct_type => |anon_struct_type| anon_struct_type.fieldName(ip, tuple_field_index),
	   .struct_type => ip.loadStructType(inst_ty.toIntern()).fieldName(ip, tuple_field_index),
	   else => unreachable,
	};
	const struct_field_index = if (maybe_field_name.unwrap()) |field_name|
            try sema.structFieldIndex(block, struct_ty, field_name, field_src)
	else
	   tuple_field_index;
```

Reply:
done